test/cpp_unit/test_lstm.cpp

```diff
@@ -0,0 +1,70 @@
+#include <gtest/gtest.h>
+
+#include "../../include/common.h"
+#include "../../include/dataloader.h"
+#include "../../include/linear_layer.h"
+#include "../../include/lstm_layer.h"
+#include "../../include/sequential.h"
+#include "../../include/slinear_layer.h"
+#include "../../include/slstm_layer.h"
+#include "test_utils.h"
+
+extern bool g_gpu_enabled;
+
+class SineSignalTest : public ::testing::Test {
+   protected:
+    void SetUp() override {}
+
+    void TearDown() override {}
+};
+
+TEST_F(SineSignalTest, LSTMTest_CPU) {
+    int input_seq_len = 4;
+    Sequential model(LSTM(1, 8, input_seq_len), LSTM(8, 8, input_seq_len),
+                     Linear(8 * input_seq_len, 1));
+    model.set_threads(2);
+    float avg_error;
+    float log_lik;
+    float mse_threshold = 0.8f;
```

Comment:
`py` and `cpp` have different threshold.  0.8 for `cpp` and 0.4 for `py`

Reply:
I set 0.5 for all tests